arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,642 @@
+      "description": "The storage account base properties.",
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
+      "required": [
+        "location",
+        "properties"
+      ],
+      "properties": {
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryPropertiesCreateParameters",
+          "description": "The parameters for creating container registry properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryPropertiesCreateParameters": {
+      "description": "The parameters for creating container registry properties.",
+      "required": [
+        "tenantId",
+        "storageAccount"
+      ],
+      "properties": {
+        "tenantId": {
+          "description": "The tenant id for the current user.",
+          "type": "string"
+        },
+        "adminUserEnabled": {
+          "description": "The boolean value that indicates whether admin user is enabled.",
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The storage account properties."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The storage account properties.",
+      "required": [
+        "name",
+        "accessKey",
+        "endPointUrl"
+      ],
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string"
+        },
+        "accessKey": {
+          "description": "Storage account access key.",
+          "type": "string"
+        },
+        "endPointUrl": {
+          "description": "Storage account endpoint URL.",
+          "type": "string"
+        },
+        "location": {
+          "description": "Storage account location.",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
+      "description": "Azure resource.",
+      "properties": {
+        "id": {
+          "description": "Resource Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string",
+          "readOnly": true

```

Comment:
@amarzavery Thanks a lot for the detailed suggestion, this is great and we will definitely adopt this model and make improvements. I just want to make sure my understanding of the scenario is correct using the same model as input/output for PUT.

GET | PUT (or even PUT | PUT), if the user modified location (since location is required and not readOnly), the RP will throw an error. We don't seem to have a way to inform the user that location cannot be changed for an existing resource, or is this considered a common sense to Azure resources?

The pipeline scenario you mentioned earlier GET | UPDATE, the user can only do this by calling a PUT after GET, if the above scenario is enabled. We can't have GET | PATCH, since PATCH uses a different model as input and all properties are optional.